sample/core.py

```diff
@@ -69,16 +69,78 @@ def get_linkshere(self, lhprop="pageid|title|redirect", lhlimit="max"):
         lh_list = {}
         res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
 
-        lh_list = _strip_JSON(res, prop, '_nothing-to-strip_')
+        lh_list = _strip_JSON(res, prop, '_nothing-to-strip_','title')
         while 'continue' in res:
             self.payload['lhcontinue'] = res['continue']['lhcontinue']
             res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
-            _append_results(lh_list, res, prop, '_nothing-to-strip_')
+            _append_results(lh_list, res, prop, '_nothing-to-strip_', 'title')
 
         self.payload['lhcontinue'] = None
         self.payload['prop'] = None
         return lh_list
 
+    # Method to fetch contributors
+
+    def get_contributors(self, pclimit="max"):
+        prop = 'contributors'
+        self.payload['prop'] = prop
+        self.payload['pclimit'] = pclimit
+
+        pc_list = {}
+        res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
+
+        pc_list = _strip_JSON(res, prop, '_nothing-to-strip_', 'name')
+        while 'continue' in res:
+            self.payload['pccontinue'] = res['continue']['pccontinue']
+            res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
+            _append_results(pc_list, res, prop, '_nothing-to-strip_', 'name')
+
+        self.payload['pccontinue'] = None
+        self.payload['prop'] = None
+        return pc_list
+
+    # Method to fetch links
+
+    def get_links(self,pllimit="max", pldir="ascending"):
+        prop = 'links'
+        self.payload['prop'] = prop
+        self.payload['pllimit'] = pllimit
+        self.payload['pldir'] = pldir
+        pl_list = {}
+        res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
+
+        pl_list = _strip_JSON(res, prop, '_nothing-to-strip_', 'title')
+        while 'continue' in res:
+            self.payload['plcontinue'] = res['continue']['plcontinue']
+            res = requests.get(self.base_url, params=self.payload,headers=self.headers).json()
```

Comment:
There are usually plugins for editors that do these checks according to PEP8. You can google if there are any for editor of your choice. For this PR I did manually.